### DISCUSSION TITLE
Large world support question

### DISCUSSION THREAD

**Discussion by OldSnake22** [google/filament#6315] [Q&A] [answered]
Hello, 

Just wanted a confirmation. Looking at the engine API it seems that large world support is already integrated in the engine ? 

I guess that shaders are still using float instead of doubles in shaders as it would be quite expensive. The large world coordinate support is then done in the math transform calculation so the transform matrixes sent to the shaders use coords relative to the camera transform or something like this ? 

Thank you in advance for your response. My project need large world coords support so thats why i'm asking :)

**Comment by romainguy** [comment, marked as answer]
Yes, large worlds are supported by moving everything relative to the camera. This is especially important for mobile where not only do we not want to use doubles, but we avoid single precision floats when possible.

@pixelflinger can explain in more details.

**Comment by OldSnake22** [comment]
Thank you very much :)
